### PR TITLE
Refactored PromRemoteWriteExporter to create client in Start to accomodate future update to ToClient() apis 

### PR DESCRIPTION
**Pull request by pavankrish123**
Description:

Updated PrometheusRemoteWriteExporter for upcoming client auth extension changes. Moved httpClient creation to Start method of the component. This ensures the call to ToClient() method gets access to host object which is required in the next version of ToClient() apis. 

Link to tracking Issue:
#3287 #3282

Testing:
Unit tests

Documentation:
Unit tests

### CODE REVIEW

**Comment by pavankrish123** [comment]
cc: @bogdandrutu @gramidt @jpkrohling

**Review by jpkrohling — APPROVED** [review]

File: exporter/prometheusremotewriteexporter/factory_test.go [hunk 1]
```diff
@@ -51,37 +52,48 @@ func Test_createMetricsExporter(t *testing.T) {
...
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			_, err := createMetricsExporter(context.Background(), tt.params, tt.cfg)
-			if tt.returnError {
+			exp, err := createMetricsExporter(context.Background(), tt.params, tt.cfg)
+			if tt.returnErrorOnCreate {
+				assert.Error(t, err)
+				return
+			}
+			assert.NotNil(t, exp)
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/factory_test.go:94]: Add an `assert.NoError(t, err)` before this null check

**Review by bogdandrutu — APPROVED** [review]

**Review by gramidt — APPROVED** [review]
After addressing the improvement Bogdan mentioned, LGTM.

File: exporter/prometheusremotewriteexporter/factory_test.go (hunk 1 above)
Comment (pavankrish123) [reply exporter/prometheusremotewriteexporter/factory_test.go:95]: Ok done!

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase the PR so you can pass CircleCI tests

**Pull request merged**